src/data/sonata_data.cpp

```diff
@@ -203,10 +205,8 @@ void SonataData::check_and_write(double timestep) {
     last_step_recorded_ += reporting_period_ * steps_recorded_;
     nodes_recorded_.clear();
 
-    // Write when buffer is full, finish all remaining recordings or when record several steps in a
-    // row
-    if (current_step_ == steps_to_write_ || current_step_ == remaining_steps_ ||
-        steps_recorded_ > 1) {
+    // Write when buffer is full or we finish all remaining recordings
+    if (current_step_ == steps_to_write_ || current_step_ == remaining_steps_) {
```

Comment:
@jorblancoa: IIRC, we had `|| steps_recorded_ > 1` condition because when simulation is finishing and if there are any steps recorded, we wanted to write those also.  Is this now covered somewhere else?

Reply:
Yes, now every time there is a recording we check if we need to write (see line 154)
When the simulation finishes there is a call to flush the data anyway

Reply:
Out of curiosity - why that call to `check_and_write` is added now every recording step?

Reply:
Ok